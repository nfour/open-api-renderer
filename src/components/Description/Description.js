import React, { PureComponent } from 'react'
import classNames from 'classnames'
import markdown from 'markdown-it'
import PropTypes from 'prop-types'
import './Description.scss'

const cm = markdown('commonmark')

export default class Description extends PureComponent {
  render () {
    const { isInline, description } = this.props
    const text = {
      __html: cm.render(description)
    }

    return (
      <div className={classNames('description', {
        'description-inline': isInline
      })} dangerouslySetInnerHTML={text} />
    )
  }
}

Description.propTypes = {
  description: PropTypes.string.isRequired,
  isInline: PropTypes.bool
}
