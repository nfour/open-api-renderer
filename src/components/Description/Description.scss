.description-inline,
.description-inline p {
  display: inline;
}
