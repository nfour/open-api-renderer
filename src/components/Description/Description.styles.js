
import { createSheet } from '../../theme'

export const styles = createSheet(({ shades }) => ({
  'description': {
    '&$inline': {
      '&, & p': {
        display: 'inline'
      }
    },
    '& p': {
      margin: '.5rem 0'
    }
  },
  'inline': {}
}))
