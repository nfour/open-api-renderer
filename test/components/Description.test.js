import React from 'react';
import Description from './../../src/components/Description/Description';
import { expect } from 'chai';
import { render } from 'enzyme';

describe('<Description />', () => {
  it('renders text with no commonmark as html', () => {
    const text = 'This method has zero markdown.';
    const wrapper = render(<Description description={text} />);

    expect(wrapper.find('p').length).to.equal(1);
  });

  it('renders commonmark text as html', () => {
    const text = 'This method has some `var i = 0` _markdown_, including a [link](http://www.google.com).';
    const wrapper = render(<Description description={text} />);

    expect(wrapper.find('p').length).to.equal(1);
    expect(wrapper.find('code').length).to.equal(1);
    expect(wrapper.find('em').length).to.equal(1);
    expect(wrapper.find('a').length).to.equal(1);
  });

  it('renders text inline if asked', () => {
    const text = 'This method has zero markdown';
    const wrapper = render(<Description isInline description={text} />);

    expect(wrapper.find('.description-inline').length).to.equal(1);
  });
});
