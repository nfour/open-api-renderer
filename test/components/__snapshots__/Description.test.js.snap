// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> renders commonmark text as html 1`] = `
<div
  className="description"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This method has some <code>var i = 0</code> <em>markdown</em>, including a <a href=\\"http://www.google.com\\">link</a>.",
    }
  }
/>
`;

exports[`<Description /> renders text inline if asked 1`] = `
<div
  className="description description-inline"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This method has zero markdown.",
    }
  }
/>
`;

exports[`<Description /> renders text with no commonmark as html 1`] = `
<div
  className="description"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This method has zero markdown.",
    }
  }
/>
`;
